cmd/controller/controller.go

```diff
@@ -405,33 +300,28 @@ func (c *Controller) updateRelease(key string) error {
 	}
 
 	log.Printf("Downloading repo %s index...", repoURL)
-	repoIndex, err := fetchRepoIndex(c.netClient, repoURL, authHeader)
+	repoIndex, err := chartUtils.FetchRepoIndex(c.netClient, repoURL, authHeader)
 	if err != nil {
 		return err
 	}
 
-	chartURL, err := findChartInRepoIndex(repoIndex, helmObj.Spec.ChartName, helmObj.Spec.Version)
-	if err != nil {
-		return err
-	}
-
-	chartURL, err = resolveChartURL(repoURL, chartURL)
+	chartURL, err := chartUtils.FindChartInRepoIndex(repoIndex, repoURL, helmObj.Spec.ChartName, helmObj.Spec.Version)
 	if err != nil {
 		return err
 	}
 
 	log.Printf("Downloading %s ...", chartURL)
-	chartRequested, err := fetchChart(c.netClient, chartURL, authHeader)
+	chartRequested, err := chartUtils.FetchChart(c.netClient, chartURL, authHeader, c.loadChart)
 	if err != nil {
 		return err
 	}
 
 	rlsName := getReleaseName(helmObj)
 	var rel *release.Release
 
-	_, err = c.helmClient.ReleaseHistory(rlsName, helm.WithMaxHistory(1))
-	if err != nil {
-		if !isNotFound(err) {
+	h, err := c.helmClient.ReleaseHistory(rlsName, helm.WithMaxHistory(1))
+	if err != nil || len(h.GetReleases()) == 0 {
```

Comment:
no, I needed to add this because the `fake` helm client doesn't return an error if the release doesn't exists. It returns an empty list. The real helm return an error so this was working properly.